File: mw.py


def merge_alternately(word1, word2):
    merged = []
    i = 0
    j = 0

    while i < len(word1) or j < len(word2):
        if i < len(word1):
            merged.append(word1[i])
            i += 1
        if j < len(word2):
            merged.append(word2[j])
            j += 1

    return ''.join(merged)


s1 = 'abc'
s2 = 'pqr'
res = merge_alternately(s1, s2)
print(res)

File: mw.go
package main

func mergeAlternatingChars(s1, s2 string) string {
	result := make([]byte, len(s1)+len(s2))
	i, j, k := 0, 0, 0

	for {
		if i < len(s1) {
			result[k] = s1[i]
			i++
			k++
		}
		if j < len(s2) {
			result[k] = s2[j]
			j++
			k++
		}
		if i >= len(s1) && j >= len(s2) {
			break
		}
	}

	return string(result[:k])
}

func main() {
	s1 := "abc"
	s2 := "pqr"

	merged := mergeAlternatingChars(s1, s2)
	print("s1: ")
	print(s1)
	print("\ns2: ")
	print(s2)
	print("\nMerged result: ")
	print(merged)
	print("\n")
}
